consensus/core/src/dag_state.rs

```diff
@@ -332,7 +333,18 @@ impl DagState {
         self.recent_blocks
             .insert(block_ref, BlockInfo::new(block.clone()));
         self.recent_refs_by_authority[block_ref.author].insert(block_ref);
-        self.threshold_clock.add_block(block_ref);
+        if self.threshold_clock.add_block(block_ref) {
+            let quorum_delay_ms = self
+                .context
+                .clock
+                .timestamp_utc_ms()
+                .saturating_sub(self.get_last_proposed_block().timestamp_ms());
+            self.context
+                .metrics
+                .node_metrics
+                .quorum_receive_latency
+                .observe(Duration::from_millis(quorum_delay_ms).as_secs_f64());
```

Comment:
Why we consider more accurate using as reference point the last proposed block time? I mean, for whatever reason s we might indeed see a round delay (ex previous leader not available yet)

Reply:
The gap between reaching quorum is the same as round interval, which can already be calculated from round rate. We need to break down round intervals further. Skipping proposals is possible but rare. I can guard this with last proposed round being the current round.